### ISSUE TITLE
add softmax function

### ISSUE THREAD

**Issue by goldingn** [greta-dev/greta#124]
Needed for building models around the categorical and multinomial distributions.

```
library (greta)
library(tensorflow)
op <- greta::.internals$nodes$constructors$op
softmax <- function (logits) {
  op("softmax", logits, tf_operation = tf$nn$softmax)
}
```

**Comment by goldingn** [comment]
For consistency with the rest of the stats-focussed interface, this will instead be implemented as `imultilogit()`, equivalent to `VGAM::multilogit(., inverse = TRUE)`, which assumes the columns of x are the *n-1* logits required to define the probabilities.
